src/plugins/custom/file-handling.ts

```diff
@@ -0,0 +1,76 @@
+import fp from 'fastify-plugin'
+import { BusboyFileStream } from '@fastify/busboy'
+import { FastifyInstance } from 'fastify'
+import path from 'path'
+import { pipeline } from 'node:stream/promises'
+import fs from 'node:fs'
+
+declare module 'fastify' {
+  export interface FastifyInstance {
+    fileHandler: {
+      upload: (fileDir: FileDir, fileName: string, file: BusboyFileStream) => Promise<void>
+      remove: (fileDir: FileDir, fileName: string) => Promise<void>
```

Comment:
```suggestion
      upload: typeof upload;
      remove: typeof remove;
```

Reply:
i can't do this, because i have to bind manually the fastify instance in the decorate function

Reply:
> i have to bind manually the fastify instance in the decorate function

Why?